develop build breaks with EXTENDED_POLICY=EXTERNAL_PROPRIETARY

### Bug Report
Since the boost lock merge, there is no `Lock(bool)` constructor. The EXTERNAL_PROPRIETARY cache_manager still has an instance of one such constructor that I missed, this should be converted to a `RecursiveLock`


##### Reproduction Steps
1. build core with EXTENDED_POLICY=EXTERNAL_PROPRIETARY
2. watch as it fails due to no appropriate constructor

##### Expected Behavior
core builds successfully with external_proprietary

##### Observed Behavior
core fails with an error message like that below

##### OS & Version Information
* OS/Version: Ubuntu 18.04
* SDL Core Version: develop
* Testing Against: just gcc

##### Test Case, Sample Code, and / or Example App
```
/home/conlain/programs/sdl_core_develop/src/components/policy/policy_external/src/cache_manager.cc:257:23: error: no matching function for call to ‘sync_primitives::Lock::Lock(bool)’
     , cache_lock_(true) {
```

--- a/src/components/policy/policy_external/include/policy/cache_manager.h
+++ b/src/components/policy/policy_external/include/policy/cache_manager.h
@@ -789,7 +789,7 @@ class CacheManager : public CacheManagerInterface {
   typedef std::set<std::string> UnpairedDevices;
   UnpairedDevices is_unpaired_;
 
-  mutable sync_primitives::Lock cache_lock_;
+  mutable sync_primitives::RecursiveLock cache_lock_;
   sync_primitives::Lock unpaired_lock_;
 
   typedef std::map<std::string, Permissions> AppCalculatedPermissions;

--- a/src/components/policy/policy_external/include/policy/update_status_manager.h
+++ b/src/components/policy/policy_external/include/policy/update_status_manager.h
@@ -227,7 +227,7 @@ class UpdateStatusManager {
 
     volatile uint32_t timeout_;
     volatile bool stop_flag_;
-    sync_primitives::Lock state_lock_;
+    sync_primitives::RecursiveLock state_lock_;
     sync_primitives::ConditionalVariable termination_condition_;
     UpdateStatusManager* update_status_manager_;
   };

--- a/src/components/policy/policy_external/src/cache_manager.cc
+++ b/src/components/policy/policy_external/src/cache_manager.cc
@@ -253,8 +253,7 @@ CacheManager::CacheManager()
     : CacheManagerInterface()
     , pt_(new policy_table::Table)
     , backup_(new SQLPTExtRepresentation())
-    , update_required(false)
-    , cache_lock_(true) {
+    , update_required(false) {
   InitBackupThread();
 }
 
@@ -262,8 +261,7 @@ CacheManager::CacheManager(bool in_memory)
     : CacheManagerInterface()
     , pt_(new policy_table::Table)
     , backup_(new SQLPTExtRepresentation(in_memory))
-    , update_required(false)
-    , cache_lock_(true) {
+    , update_required(false) {
   InitBackupThread();
 }
 

--- a/src/components/policy/policy_external/src/update_status_manager.cc
+++ b/src/components/policy/policy_external/src/update_status_manager.cc
@@ -209,7 +209,6 @@ UpdateStatusManager::UpdateThreadDelegate::UpdateThreadDelegate(
     UpdateStatusManager* update_status_manager)
     : timeout_(0)
     , stop_flag_(false)
-    , state_lock_(true)
     , update_status_manager_(update_status_manager) {
   LOG4CXX_AUTO_TRACE(logger_);
   LOG4CXX_DEBUG(logger_, "Create UpdateThreadDelegate");

